strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/layout/npm/NpmArtifactController.java

```diff
@@ -346,6 +355,14 @@ public void downloadPackage(@RepositoryMapping Repository repository,
         final String storageId = repository.getStorage().getId();
         final String repositoryId = repository.getId();
 
+        if (!packageNameWithVersion.startsWith(packageName + "-"))
+        {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        }
+
+        //Example of packageNameWithVersion  core-9.0.1-next.8.tgz
```

Comment:
This comment would probably be better in the test.

Reply:
This comment is specific to SpringMVC mapping. That is why I mentioned it here. If these comments are not to be mentioned in the classes, I can remove them completely.

Reply:
Ah, okay! Leave it then. It doesn't hurt! Thanks for clarifying! :)